### PR TITLE
refactor: import oracles

### PR DESCRIPTION
**Pull request by 0xsambugs**
We are now importing and using the new oracle definition from the oracles repository

_Note: I'm sorry that I had to modify so many files, but I couldn't make this smaller_

### CODE REVIEW

File: contracts/DCAHub/DCAHubSwapHandler.sol [hunk 1]
```diff
@@ -114,9 +114,9 @@ abstract contract DCAHubSwapHandler is ReentrancyGuard, DCAHubConfigHandler, IDC
     address _tokenB,
     uint256 _magnitudeA,
     uint256 _magnitudeB,
-    IPriceOracle _oracle
+    ITokenPriceOracle _oracle
   ) internal view virtual returns (uint256 _ratioAToB, uint256 _ratioBToA) {
-    _ratioBToA = _oracle.quote(_tokenB, uint128(_magnitudeB), _tokenA);
```
Comment (0xsambugs) [inline contracts/DCAHub/DCAHubSwapHandler.sol:119]: I love that we can get rid of this <3

File: package.json [hunk 2]
```diff
@@ -61,59 +61,64 @@
...
     "@chainlink/contracts": "0.2.2",
     "@mean-finance/deterministic-factory": "1.3.1",
     "@mean-finance/nft-descriptors": "1.0.0-rc3",
+    "@mean-finance/oracles": "2.0.0-rc3",
     "@openzeppelin/contracts": "4.7.1",
     "@uniswap/v3-periphery": "1.2.0",
     "base64-sol": "1.0.1"
   },
   "devDependencies": {
-    "@0xged/hardhat-deploy": "0.11.5",
```
Comment (0xsambugs) [inline package.json:72]: Copied dependencies from other repos

File: test/integration/oracles/chainlink-oracle.spec.ts [hunk 3]
```diff
@@ -82,7 +82,7 @@ const PLANS: { tokenIn: Token; tokenOut: Token }[][] = [
 
 const TRESHOLD_PERCENTAGE = 2.5; // In mainnet, max threshold is usually 2%, but since we are combining pairs, it can sometimes be a little higher
 
-contract('ChainlinkOracle', () => {
+contract.skip('ChainlinkOracle', () => {
```
Comment (0xsambugs) [inline test/integration/oracles/chainlink-oracle.spec.ts:85]: Will delete this test in the following PR

File: test/integration/oracles/uniswap-v3-oracle.spec.ts [hunk 4]
```diff
@@ -18,11 +18,10 @@ const UNI_WETH_USDC_POOL_LOW = '0x88e6A0c2dDD26FEEb64F039a2c41296FcB3f5640';
 const UNI_WETH_USDC_POOL_MEDIUM = '0x8ad599c3A0ff1De082011EFDDc58f1908eb6e6D8';
 const UNI_WETH_USDC_POOL_HIGH = '0x7BeA39867e4169DBe237d55C8242a8f2fcDcc387';
 
-contract('UniswapV3Oracle', () => {
+contract.skip('UniswapV3Oracle', () => {
```
Comment (0xsambugs) [inline test/integration/oracles/uniswap-v3-oracle.spec.ts:21]: Will delete this test in the following PR

File: test/unit/DCAHub/dca-hub-position-handler.spec.ts [hunk 5]
```diff
@@ -688,10 +688,12 @@ contract('DCAPositionHandler', () => {
         const withdrawer = await readArgFromEventOrFail(response, 'WithdrewMany', 'withdrawer');
         const withdrawRecipient = await readArgFromEventOrFail(response, 'WithdrewMany', 'recipient');
         const positions = await readArgFromEventOrFail<any>(response, 'WithdrewMany', 'positions');
-        const withdrew = await readArgFromEventOrFail(response, 'WithdrewMany', 'withdrew');
+        const withdrew = await readArgFromEventOrFail<BigNumber[]>(response, 'WithdrewMany', 'withdrew');
         expect(withdrawer).to.equal(owner.address);
         expect(withdrawRecipient).to.equal(recipient);
-        expect(withdrew).to.eql([constants.ZERO, constants.ZERO]);
```
Comment (0xsambugs) [inline test/unit/DCAHub/dca-hub-position-handler.spec.ts:694]: Not sure why this doesn't work anymore. Probably has to do with the dependencies updates

Comment (0xged) [reply test/unit/DCAHub/dca-hub-position-handler.spec.ts:694]: Hm... That sucks. Yeah, it's a dependencies update, for sure, not ours.

**Review by 0xged — APPROVED** [review]
Looks good! A lot of files, but pretty straight forward changes 😊

**Pull request merged**